hls/lib/hls_endpoint.ex

```diff
@@ -156,11 +164,14 @@ defmodule Membrane.RTC.Engine.Endpoint.HLS do
       |> Enum.map(&{&1, track_id})
       |> Enum.filter(&Map.has_key?(ctx.children, &1))
 
-    {removed_track, tracks} = Map.pop!(state.tracks, track_id)
-    state = %{state | tracks: tracks}
+    {removed_track, subscriber} = Subscriber.pop_track!(state.subscriber, track_id)
```

Comment:
What do you think about one function `Subscriber.remove_track` that would return tuple `{removed_track, subscriber}`.